File: backend/src/main/java/com/example/backend/Model/Class/ProcessLog.java
package com.example.backend.Model.Class;

import com.example.backend.Model.Enum.ProcessLogStatus;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.cglib.core.Local;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Data
@Entity
@Cacheable
@Table(name = "PROCESS_LOG")
public class ProcessLog {
    @Id
    private String id;
    @Column(name = "start_time")
    private LocalDateTime start_time;
    @Column(name = "end_time")
    private LocalDateTime end_time;
    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private ProcessLogStatus status;
    @ManyToOne
    @JoinColumn(name = "process_id", referencedColumnName = "id")
    private Process process_id;
    @ManyToOne
    @JoinColumn(name ="employee_id", referencedColumnName = "id")
    private Employees employee_id;
    @ManyToOne
    @JoinColumn(name = "car_id", referencedColumnName = "id")
    private Cars car_id;
    @ManyToOne
    @JoinColumn(name = "machine_id", referencedColumnName = "id")
    private Machines machine_id;
}

File: frontend/src/app/_model/_interface/process-log.ts
import { Cars } from './car';
import { Employee } from './employee';
import { Machines } from './machine';
import { Process } from './process';

export interface ProcessLog {
  id: string;
  start_time: string;
  end_time: string;
  status: string;
  process_id: Process;
  employee_id: Employee;
  car_id: Cars;
  machine_id: Machines;
}

export function isProcessLog(obj: any): obj is ProcessLog {
  return (
    !!obj &&
    typeof obj === 'object' &&
    'machine_id' in obj &&
    obj.machine_id &&
    typeof obj.machine_id.id === 'string'
  );
}
